src/Cns.ts

```diff
@@ -18,46 +15,27 @@ import {
 import { isValidTwitterSignature } from './utils/TwitterSignatureValidator';
 import NamingService from './NamingService';
 import NetworkConfig from './config/network-config.json';
+import ConfigurationError, { ConfigurationErrorCode } from './errors/configurationError';
 
 export default class Cns extends EthereumNamingService {
   readonly contract: Contract;
-  reader: ICnsReader;
   static TwitterVerificationAddress = '0x12cfb13522F13a78b650a8bCbFCf50b7CB899d82';
   static ProxyReaderMap: ProxyReaderMap = getProxyReaderMap();
 
   constructor(source: SourceDefinition = {}) {
     super(source, NamingServiceName.CNS);
-    this.contract = this.buildContract(proxyReaderAbi, this.registryAddress!);
-  }
-
-  async getReader(): Promise<ICnsReader> {
-    if (!this.reader) {
-      this.reader = (await this.isDataReaderSupported())
-        ? new CnsProxyReader(this.contract)
-        : new CnsRegistryReader(this.contract);
+    const network = Number(source.network) || 1;
+    const proxyContractAddress  = this.defaultRegistry(network);
```

Comment:
`source.network` is unnormalized data. Take normalized data from `this.network`:

```suggestion
    const proxyContractAddress  = this.defaultRegistry(this.network);
```